extensions/ANGLE/EGL_ANGLE_sync_control_rate.txt

```diff
@@ -0,0 +1,110 @@
+﻿Name
+
+   ANGLE_sync_control_rate
+
+Name Strings
+
+   EGL_ANGLE_sync_control_rate
+
+Contact
+
+   Jonah Ryan-Davis, Google (jonahr 'at' google.com)
+
+Status
+
+   Draft.
+
+Version
+
+   Version 1, 2020-03-24
+
+   Based on GLX_OML_sync_control Revision 6.0
+
+Number
+
+   ???
+
+Dependencies
+
+   The extension is written against the EGL 1.2 Specification, although it
+   should work on other versions of these specifications. This extension
+   also requires an operating system which supports CLOCK_MONOTONIC.
+
+Overview
+
+   This extension provides counters which let applications know about the
+   timing of the last vertical retrace. By looking at the system clock, as
+   well as the refresh rate of the monitor, this should enable applications
+   to predict the position of future retraces so as to schedule an optimal
+   workload.
+
+   This extension incorporates the use of three counters that provide
+   the necessary synchronization. The Unadjusted System Time (or UST)
+   is the 64-bit CLOCK_MONOTONIC clock; in particular this lets the
+   application schedule future vertical retraces by querying this clock.
+   The graphics Media Stream Counter (or graphics MSC) is a counter
+   that is unique to the graphics subsystem and increments for each
+   vertical retrace that occurs. The Swap Buffer Counter (SBC) is an
+   attribute of an EGLSurface and is incremented each time a swap
+   buffer action is performed on the associated surface.
```

Comment:
This graf seems like it copies too much from the GLX spec, you don't actually define a way to get at the SBC from this spec and unless I'm missing something I don't see any other extension that would expose it.

I'm not sure you really want to _specify_ that the UST is `clock_gettime(CLOCK_MONOTONIC)`, it's a fine choice for POSIXy systems but I'm not sure what you'd want that to map to on Windows, Fuschia, etc. Might be better to leave that to an Implementation Notes section.

Reply:
hmm, SBC is probably from the EGL_CHROMIUM_sync_control extension? (which is implemented for example in Mesa since 2014, but the extension spec is not in the registry for some reason, open issue here: https://github.com/KhronosGroup/EGL-Registry/issues/115 and the extension spec here: https://chromium.googlesource.com/chromium/src/gpu/+/master/GLES2/extensions/CHROMIUM/EGL_CHROMIUM_sync_control.txt )

Reply:
@jonahryandavis can you confirm this?

Reply:
You're right, this description was a mistake because originally we had simply added this function to the existing EGL_CHROMIUM_sync_control, but split out getMSCRate into this extension shortly after. I've cleaned it up, let me know if you have other suggestions, sorry for the delay!